streaming/vision.py

```diff
@@ -0,0 +1,154 @@
+# Copyright 2023 MosaicML Streaming authors
+# SPDX-License-Identifier: Apache-2.0
+
```

Comment:
Feels like this file does fit well here!

Reply:
+1, I agree. Are people using these classes? if so, can we put this back in a `streaming/vision` directory or something?

Reply:
Hmmm.

How much torchvisioning have you done? https://pytorch.org/vision/main/_modules/index.html :eyes: They're basically all `VisionDataset`s -> this is a standard PyTorch idiom and is like 1/3 as basey as `StreamingDataset`, which is basey enough for me :)

And recall the time RR had to key off some composer behavior on whether a streaming dataset was vision.

But yeah, we should only keep base classes in `streaming/`, so we need to delete `StreamingDataset` because it's not the base class, it descends from `Array` along with `LocalDataset`, to which we may add more subclasses in the future for different use cases (c.f. that user request about a lazy map-style non-streaming Streaming dataset, etc).

Also these files are pretty diverse already, I don't think a (modality).py sticks out?

I totally see moving the Streaming-specific innovation `convert_image_class_dataset()` out to a `examples/vision/base.py`, but the rest is a la [torchvision/datasets/vision.py](https://pytorch.org/vision/main/_modules/torchvision/datasets/vision.html), let's keep it in a `streaming/vision.py` I think?

```
array.py
batching/
compression.py
constant.py
converters/
dataloader.py
dataset.py
distributed.py
format/
hashing.py
__init__.py
local.py
partition/
py.typed
sampling.py
shared/
shuffle/
spanner.py
storage/
stream.py
util.py
_version.py
vision.py
world.py
```